### ISSUE TITLE
looking to create ja3

### ISSUE THREAD

**Issue by kolet** [salesforce/ja3#83]
hi there , i'd like to know if there is any tools to create ja3 finger prints . 

all of the tools i've seen till now , are tools that are extracting ja3 finger prints and their md5 hash.

however i am looking for a tool that create "generate" those ja3 finger prints .

thanks

**Comment by 89z** [comment]
I don't understand the question. Can you explain what you are trying to do?

**Comment by kolet** [comment]
> I don't understand the question. Can you explain what you are trying to do?

its more like a question not an issue .

i would like to generate ja3 fingerprints , and i am looking a script/tool to do that .

**Comment by 89z** [comment]
> i would like to generate ja3 fingerprints 

yeah, you said that already. What you're asking doesn't make sense. Fingerprint is just a string 32 bytes long:

~~~
9fc6ef6efc99b933c5e2d8fcf4f68955
~~~

using hex characters. So just write a program

~~~go
package main

import (
   "crypto/md5"
   "encoding/hex"
   "fmt"
)

func main() {
   hash := md5.New()
   var buf [1]byte
   for i := 0; i < 9; i++ {
      hash.Write(buf[:])
      fmt.Println(hex.EncodeToString(hash.Sum(nil)))
   }
}
~~~

result:

~~~
93b885adfe0da089cdf634904fd59f71
c4103f122d27677c9db144cae1394a66
693e9af84d3dfcc71e640e005bdc5e2e
f1d3ff8443297732862df21dc4e57262
ca9c491ac66b2c62500882e93f3719a8
7319468847d7b1aee40dbf5dd963c999
d310a40483f9399dd7ed1712e0fdd702
7dea362b3fac8e00956a4952a3d4f474
3f2829b2ffe8434d67f98a2a98968652
~~~

**Comment by kolet** [comment]
im not sure how i am not making sense :) 


md5 hashes are not what i am looking for ..
valid fingerprints are .


i am already extracting all ja3 from my network through suricata .

i'd like to generate ja3 constantly untill i stop the script for example .

i tried example.go (from UTLS ), however it generate mots of the time the same JA3 .

what i am looking for a generating hundreds - thousands of ja3 fingerprints .

**Comment by 89z** [comment]
sounds like you're pretty clearly looking to abuse some server somewhere, with this knowledge. I am not interested to help with that *pro bono*. If you wish to discuss business, my contact info is in my bio. Otherwise good luck to you.

**Comment by kolet** [comment]
> 

this is exactly what i am trying to do..
i'd like to avoid being blocked while scraping .

**Comment by 89z** [comment]
I cannot help with this *pro bono*. If you want to do business, my contact info is in my bio. Otherwise good luck.

**Comment by kolet** [comment]
> I cannot help with this _pro bono_. If you want to do business, my contact info is in my bio. Otherwise good luck.
thanks !

hopefully some one will come and help out , its github after all :) 

they are companies that does that (scrapify for example ), i try to avoid that .

**Comment by wwhtrbbtt** [comment]
You shouldn't randomize your ja3. Use the one that chrome uses, and stick with it. That will work way better

**Comment by aygupt1822** [comment]
> hi there , i'd like to know if there is any tools to create ja3 finger prints .
> 
> all of the tools i've seen till now , are tools that are extracting ja3 finger prints and their md5 hash.
> 
> however i am looking for a tool that create "generate" those ja3 finger prints .
> 
> thanks

Oh boiiiiii, Well this can be done in the python with random Ciphers, EC and EC_PF.
Automation can be done on this. 

But you need to first select the Ciphers suites, EC and EC_PF for this purpose, and then randomize between them.
